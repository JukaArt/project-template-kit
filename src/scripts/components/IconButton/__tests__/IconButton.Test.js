import React, { Component } from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import IconButton from 'Components/IconButton';

describe('Button test', () => {

  const props = {
    className:'IconNode',
    icon: 'http://example.com/icon.svg',
    top: '10px',
    left: '10px',
    onClick: jest.fn()
  }

  const iconButton = shallow(<IconButton {...props} />);

  it('should exist ', () => {
    expect(iconButton.exists()).toBe(true);
  });

  it('Should be a structure', () => {    
    const tree = renderer.create(
      <IconButton {...props} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })
})