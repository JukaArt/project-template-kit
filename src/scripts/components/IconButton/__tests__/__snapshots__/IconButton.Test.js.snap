// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test Should be a structure 1`] = `
<div
  className="IconNode icon-http://example.com/icon.svg root"
  onClick={[Function]}
  style={
    Object {
      "color": undefined,
      "left": "10px",
      "right": undefined,
      "top": "10px",
      "zIndex": undefined,
    }
  }
/>
`;
